pkg/apis/projectcalico/v3/calico/node.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2017,2020-2021 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package calico
+
+import (
+	"github.com/projectcalico/api/pkg/lib/numorstring"
+)
+
+const (
+	KindNode     = "Node"
+	KindNodeList = "NodeList"
+	CalicoNodeIP = "CalicoNodeIP"
+	InternalIP   = "InternalIP"
+	ExternalIP   = "ExternalIP"
+)
+
+// NodeSpec contains the specification for a Node resource.
+type NodeSpec struct {
```

Comment:
oh this is an interesting one. I need to think about this more - the `projectcalico.org/v3 Node` resource is not available via the API server - it's a pseudo resource that only exists via calicoctl today.

I wonder if this particular resource actually belongs here... will think about it.